app/lib/screens/signing/sign_with_link.dart

```diff
@@ -0,0 +1,363 @@
+import 'package:threebotlogin/services/signing_service.dart';
+import 'package:threebotlogin/widgets/custom_dialog.dart';
+
+class SignWithLinkScreen extends ConsumerStatefulWidget {
+  const SignWithLinkScreen({super.key});
+
+  @override
+  ConsumerState<SignWithLinkScreen> createState() => _SignWithTextScreenState();
+}
+
+class _SignWithTextScreenState extends ConsumerState<SignWithLinkScreen> {
+  final TextEditingController _textController = TextEditingController();
+  Wallet? selectedWallet;
+  String? signedData;
+  bool isLoading = false;
+  final TextEditingController _dataController = TextEditingController();
+  final TextEditingController _linkController = TextEditingController();
+  String? linkError;
+  String? dataError;
+  String? walletError;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _textController.dispose();
+    super.dispose();
+  }
+
+  void _validateInputs() {
+    setState(() {
+      if (_linkController.text.isEmpty) {
+        linkError = 'Please enter a link';
+      } else {
+        try {
+          Uri.parse(_linkController.text);
+          linkError = null;
+        } catch (e) {
+          linkError = 'Invalid link format';
+        }
+      }
+
+      if (_dataController.text.isEmpty) {
+        dataError = 'No data extracted from link';
+      } else {
+        dataError = null;
+      }
+
+      if (selectedWallet == null) {
+        walletError = 'Please select a wallet';
+      } else {
+        walletError = null;
+      }
+    });
+  }
+
+  void _showInvalidLinkDialog() {
+    showDialog(
+      context: context,
+      builder: (BuildContext context) => CustomDialog(
+        type: DialogType.Error,
+        image: Icons.error,
+        title: 'Invalid Link',
+        description: 'The link is missing required information or is invalid.',
+        actions: [
+          TextButton(
+            child: const Text('Close'),
+            onPressed: () {
+              Navigator.pop(context);
+            },
+          ),
+        ],
+      ),
+    );
+  }
+
+  void _processLink() {
+    if (_linkController.text.isEmpty) {
+      setState(() {
+        linkError = 'Please enter a link';
+      });
+      return;
+    }
+
+    try {
+      final Uri link = Uri.parse(_linkController.text);
+      Map<String, String> queryParams = link.queryParameters;
+
+      List<String> requiredParams = [
+        'dataHash',
+        'state',
+        'appId',
+        'dataUrl',
+        'isJson',
+        'friendlyName'
+      ];
+
+      bool isValidSignAttempt = true;
+      for (var param in requiredParams) {
+        if (queryParams[param] == null || queryParams[param] == 'undefined') {
+          isValidSignAttempt = false;
+          break;
+        }
+      }
+
+      if (!isValidSignAttempt) {
+        setState(() {
+          linkError = 'Missing required parameters';
+        });
+        _showInvalidLinkDialog();
```

Comment:
tried to use the raw content of the repo readme as the link but it fails 
https://raw.githubusercontent.com/threefoldtech/tfgrid-sdk-dart/refs/heads/development/README.md

<img src="https://github.com/user-attachments/assets/74aec1d6-a31a-422b-bcb5-443fc4232135" width=250>

Reply:
fixed 
<img src="https://github.com/user-attachments/assets/affa28f6-7a50-4af7-95c0-10ef0f5813c4" width=200>